test.py

```diff
@@ -1,39 +1,262 @@
 #!/usr/bin/env python3
 
-from parse import *
+from unittest.mock import patch, mock_open
+import parse
 import logging
 import unittest
+import constants
+import csv
 
-class EncodingLineTest(unittest.TestCase):
-    def setUp(self):
-        logger = logging.getLogger()
-        logger.disabled = True
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
-    def assertError(self, string):
-        self.assertRaises(SystemExit, process_enc_line, string, 'rv_i')
-
-    def test_lui(self):
-        name, data = process_enc_line('lui     rd imm20 6..2=0x0D 1=1 0=1', 'rv_i')
+class TestProcessEncLine(unittest.TestCase):
+    @patch('builtins.open', new_callable=mock_open, read_data='lui rd 6..2=0x0D 1=1 0=1')
+    def test_process_enc_line(self, mock_file):
+        """Test the process_enc_line function."""
+        name, result = parse.process_enc_line('lui rd 6..2=0x0D 1=1 0=1', 'rv_i')
         self.assertEqual(name, 'lui')
-        self.assertEqual(data['extension'], ['rv_i'])
-        self.assertEqual(data['match'], '0x37')
-        self.assertEqual(data['mask'], '0x7f')
+        self.assertIn('encoding', result)
+        self.assertIn('variable_fields', result)
+        self.assertIn('extension', result)
+        self.assertIn('match', result)
+        self.assertIn('mask', result)
+
+class TestSameBaseISA(unittest.TestCase):
+    def test_same_base_isa(self):
+        """Test the same_base_isa function."""
+        self.assertTrue(parse.same_base_isa('rv32', ['rv32_i', 'rv64']))
+        self.assertFalse(parse.same_base_isa('rv32', ['rv64']))
+
+class TestOverlapFunctions(unittest.TestCase):
+    def test_overlaps(self):
+        """Test the overlaps function."""
+        self.assertTrue(parse.overlaps('1100', '1---'))
+        self.assertFalse(parse.overlaps('1100', '0011'))
+
+class TestCreateInstDict(unittest.TestCase):
+    def test_create_inst_dict(self):
+        """Test the create_inst_dict function."""
+        with patch('glob.glob', return_value=['mock_file.rv32_i']):
+            with patch('builtins.open', new_callable=mock_open, read_data='lui rd 6..2=0x0D 1=1 0=1'):
```

Comment:
As suggested earlier in my comments about `lui rd 6..2=0x0D 1=1 0=1` to move to a global variable, we can see the usefulness here. We can just refer that test data at multiple places. This is a best approach and also more readable. Additionally, it will help for quick fix and modifications at later stages.